kolibri/plugins/slideshow_render/assets/src/views/SlideshowRendererComponent.vue

```diff
@@ -0,0 +1,214 @@
+<template>
+
+  <CoreFullscreen
+    ref="slideshowRenderer"
+    class="slideshow-renderer"
+    @changeFullscreen="isInFullscreen = $event"
+  >
+    <UiIconButton
+      class="btn"
+      :ariaLabel="isInFullscreen ? $tr('exitFullscreen') : $tr('enterFullscreen')"
+      color="primary"
+      size="small"
+      @click="$refs.slideshowRenderer.toggleFullscreen()"
+    >
+      <mat-svg v-if="isInFullscreen" name="fullscreen_exit" category="navigation" />
+      <mat-svg v-else name="fullscreen" category="navigation" />
+    </UiIconButton>
+    <Hooper v-if="slides" @slide="handleSlide">
+      <Slide v-for="slide in slides" :key="slide.id">
+        <div
+          class="slideshow-slide-image"
+          :style="{
+            height: `calc(100% - ${caption_height}px)`
+          }"
+        >
+          <img
+            :src="slide.storage_url"
+            :aria-labelledby="slideTextId(slide.id)"
+          >
+        </div>
+        <div :id="slideTextId(slide.id)" class="hidden-descriptive-text">
+          {{ slide.descriptive_text || slide.caption }}
+        </div>
+        <div :ref="slide.id" class="caption">
+          {{ slide.caption }}
+        </div>
+      </Slide>
+      <HooperNavigation
+        slot="hooper-addons"
+        :class="{'hooper-navigation-fullscreen' : isInFullscreen}"
+      />
+      <HooperPagination slot="hooper-addons" />
+    </Hooper>
+  </CoreFullscreen>
+
+</template>
+
+
+<script>
+
+  import _ from 'lodash';
+
+  import themeMixin from 'kolibri.coreVue.mixins.themeMixin';
+  import contentRendererMixin from 'kolibri.coreVue.mixins.contentRendererMixin';
+  import responsiveElement from 'kolibri.coreVue.mixins.responsiveElement';
```

Comment:
when possible it's nice to avoid reliance on `responsiveElement` but sometimes it's a necessary evil

Reply:
Noted! 👍 Thanks Devon